plugins/woocommerce/includes/class-wc-mobile-messaging-handler.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Prepares formatted mobile deep link navigation link for order mails
+ *
+ * @package WooCommerce\Emails
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Mobile_Messaging_Handler
+ */
+class WC_Mobile_Messaging_Handler {
+
+	const OPEN_ORDER_INTERVAL_DAYS = 30;
+
+	/**
+	 * Prepares mobile messaging with a deep link
+	 *
+	 * @param WC_Order $order order that mobile message is created for.
+	 * @param ?int     $blog_id  of blog to make a deep link for.
+	 * @param DateTime $now      current DateTime.
+	 *
+	 * @return string|null
+	 */
+	public static function prepare_mobile_message(
+		WC_Order $order,
+		?int $blog_id,
+		DateTime $now
+	): ?string {
+		try {
+			$last_mobile_used = self::get_closer_mobile_usage_date();
+
+			$used_app_in_last_month = $last_mobile_used->diff( $now )->days <= self::OPEN_ORDER_INTERVAL_DAYS;
+			$has_jetpack            = null !== $blog_id;
+
+			if ( $used_app_in_last_month && $has_jetpack ) {
+				if ( self::is_store_in_person_payment_eligible() ) {
+					if ( self::is_order_in_person_payment_eligible( $order ) ) {
+						return self::accept_payment_message( $blog_id, $order->get_id() );
+					} else {
+						return self::manage_order_message( $blog_id, $order->get_id() );
+					}
+				} else {
+					return self::manage_order_message( $blog_id, $order->get_id() );
+				}
+			} else {
+				return self::no_app_message();
+			}
+		} catch ( Exception $e ) {
+			return null;
+		}
+	}
+
+	/**
+	 * Returns if store is eligible to accept In-Person Payments
+	 *
+	 * @return true if store is eligible, false otherwise
+	 */
+	private static function is_store_in_person_payment_eligible(): bool {
+		return self::has_store_specified_country_currency( 'US', 'USD' ) || self::has_store_specified_country_currency( 'CA', 'CAD' );
+	}
+
+	/**
+	 * Returns if order is eligible to accept In-Person Payments
+	 *
+	 * @param WC_Order $order order that the conditions are checked for.
+	 *
+	 * @return true if order is eligible, false otherwise
+	 */
+	private static function is_order_in_person_payment_eligible( WC_Order $order ): bool {
+		return in_array( $order->get_status(), array( 'pending', 'on-hold', 'processing' ), true ) &&
+			   in_array( $order->get_payment_method(), array( 'cod', 'woocommerce_payments', 'none' ), true ) && // phpcs:ignore WordPress.WhiteSpace.PrecisionAlignment.Found
+			   self::is_store_in_person_payment_eligible() && // phpcs:ignore WordPress.WhiteSpace.PrecisionAlignment.Found
```

Comment:
This condition is duplicated (at this point we already checked if store is IPP eligible) but I find this check needed to make sure that `is_order_in_person_payment_eligible` function does exactly what it says it does - checks all conditions.

I believe it's so minor that not even measurable in terms of performance so decided to leave it.

Reply:
(See my other comment on the next line down)